compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -131,3 +132,41 @@ private val truePredicate: (Any?) -> Boolean = { true }
  * method throws an exception if more than one element is found.
  */
 internal fun <T> Iterable<T>.singleOrEmpty(): T? = singleOrEmpty(truePredicate)
+
+internal fun TypeName.toStringOfSimpleNamesOnly(): String {
+  return removePackageNames(toString())
+}
```

Comment:
This extension (and the subsequent String parsing function) ignore all of the work that `TypeName` has already achieved.

If the type name has simple names, then it's a `ClassName` or a `ParameterizedTypeName`.  You could do something like:

```kotlin
internal fun TypeName.toStringOfSimpleNamesOnly(): String {
  return when (this) {
    is ParameterizedTypeName -> rawType.toStringOfSimpleNamesOnly()
    is ClassName -> simpleNames.joinToString(separator = ".")
    is LambdaTypeName,
    is TypeVariableName,
    is WildcardTypeName,
    Dynamic,
    -> error("???")
  }
}
```

From the context in which this is being called, the type name should be one of the first two.  If I was the one creating this extension, it would be inside the `BindsMethodValidator` class.

Reply:
Thanks, I see! But this part is no longer needed since I've removed simple names parsing.